pppp.c

```diff
@@ -497,16 +534,44 @@ int startup()
 	printf("HTTP connection from %s:%d\n",
 		inet_ntoa(iclientip->sin_addr), ntohs(iclientip->sin_port));
 	while (1) {
-		len = read(client, inbuf, sizeof(inbuf));
-		if (len < 7) {
-			close(client);
-			goto again;
-		}
-		if (strncasecmp(inbuf, "GET ", 4)) {
-			close(client);
-			goto again;
-		}
-		if (!strncmp(inbuf+4, "/v.mjpg ", 8)) {
+        char inbuf[1024];
+        int len = read(client, inbuf, sizeof(inbuf));
+        if (len <= 7) {
+            close(client);
+            goto again;
+        }
+		// if (strncasecmp(inbuf, "GET ", 4)) {
+		// 	close(client);
+		// 	goto again;
+		// }
+        if (!strncmp(inbuf+4, "/exec ", 7)) {
+            char *json_end = strstr(inbuf+7, "\r\n");
+            if (json_end) {
+                *json_end = '\0'; // Null terminate the JSON string
+                sendExecCmd(inbuf+7);
+            } else {
+                fprintf(stderr, "Invalid JSON format\n");
+            }
+    		// Send HTTP status line
+			char status_line[256];
+			sprintf(status_line, "HTTP/1.1 200 OK\r\n");
```

Comment:
sprintf returns the length of its output, so you should save it and use it, instead of using strlen(). Again, in well written code you should very rarely ever need strlen().